### ISSUE TITLE
Plug-in starts but alignments are not computed

### ISSUE THREAD

**Issue by matheusPereiraAlmeida** [giacomolanciano/PlanningBasedAlignment#1]
I am trying to use the tool to reproduce the results in the article **A Tool for Aligning Event Logs and Prescriptive Process
Models through Automated Planning**. I followed the instructions in the ReadMe.txt and I'm using the data available in the following link https://bitbucket.org/And182/plan-based-imperative-conformance-checker/src/master/resources/Petri%20Nets/Syntetic%20Petri%20Nets/d31.pnml

I have all the dependencies working correctly
<img width="293" alt="image1" src="https://user-images.githubusercontent.com/61329801/142688107-f2d5c7ed-a17c-4fce-8a04-aa75a831d65d.PNG">

I'm using the d31.pnml and the d31.xes in the tool, below are some images of the tool

![image](https://user-images.githubusercontent.com/61329801/142688351-9835c0fb-2ed1-4e3b-9ef2-6720e68a84e8.png)

![image](https://user-images.githubusercontent.com/61329801/142688379-0ba533a6-b809-456d-a348-d2d9c69d50c0.png)

![image](https://user-images.githubusercontent.com/61329801/142688395-f76114da-401f-4ce4-aa84-794d9aca25b3.png)

![image](https://user-images.githubusercontent.com/61329801/142688419-85153389-ca33-4f24-ae56-453686197a52.png)

![image](https://user-images.githubusercontent.com/61329801/142688546-91bd6e27-74aa-4bff-9614-7ee32c95fb87.png)

![image](https://user-images.githubusercontent.com/61329801/142688592-4d4f08fc-44d7-4846-abcb-f6ec6bc2d9d9.png)

![image](https://user-images.githubusercontent.com/61329801/142688625-ceca5598-c13e-4852-9257-b5aeabc41e86.png)

![image](https://user-images.githubusercontent.com/61329801/142688657-e86fd1f5-6d38-432a-95b9-3d831d8b2527.png)

 I also runned the plugin Generate PDDL Encoding for Planning-based Alignment and I got the following results
[pddl_files_1637354301319.zip](https://github.com/giacomolanciano/PlanningBasedAlignment/files/7572822/pddl_files_1637354301319.zip)

**Comment by matheusPereiraAlmeida** [comment]
I just runned the result from the log in this link http://planning.domains and I got the following error
![image](https://user-images.githubusercontent.com/61329801/142696883-bbf3b951-451b-49cc-bbbd-e65f9b8b9ee3.png)

**Comment by giacomolanciano** [comment]
@matheusPereiraAlmeida 
TL;DR: I had been too optimistic in thinking that any Python `3.2+` would have been good to run the plug-in. :-)

I am not sure which is the actual version of the code you are using. Most likely, you got the latest release from the nightly builds, that fetch the source from the [official SVN repo](https://svn.win.tue.nl/repos/prom/Packages/PlanningBasedAlignment/). In any case, the `master` branch of this repo is now again in sync with that, so I am assuming this version of the code in what follows.

I was able to replicate your error state by using Python `3.10` (as you did). In particular, I got this error message the moment the plug-in tried to start the underlying Fast Downward process:

```
Traceback (most recent call last):
  File "<REDACTED>\\PlanningBasedAlignment\\fast-downward\\builds\\release64\\bin\\translate\\translate.py", line 690, in <module>
  main()
  File "<REDACTED>\\PlanningBasedAlignment\\fast-downward\\builds\\release64\\bin\\translate\\translate.py", line 680, in main
    sas_task = pddl_to_sas(task)
	File "<REDACTED>\\PlanningBasedAlignment\\fast-downward\\builds\\release64\\bin\\translate\\translate.py", line 532, in pddl_to_sas
    groups, mutex_groups, translation_key = fact_groups.compute_groups(
	File "<REDACTED>\\PlanningBasedAlignment\\fast-downward\\builds\\release64\\bin\\translate\\fact_groups.py", line 110, in compute_groups
    groups = invariant_finder.get_groups(task, reachable_action_params)
	File "<REDACTED>\\PlanningBasedAlignment\\fast-downward\\builds\\release64\\bin\\translate\\invariant_finder.py", line 132, in get_groups
    invariants = sorted(find_invariants(task, reachable_action_params))
	File "<REDACTED>\\PlanningBasedAlignment\\fast-downward\\builds\\release64\\bin\\translate\\invariant_finder.py", line 100, in find_invariants
    start_time = time.clock()
	AttributeError: module \'time\' has no attribute \'clock\'
```

However, by switching to Python `3.2` (the version I used the last time I worked on this project) I was able to get the resulting alignments. You can tell the plug-in is actually working when you can see this progress bar (beware that the whole process may take many time for large logs and models):

![alignment-in-progress](https://user-images.githubusercontent.com/13543778/143697816-13b4e2b4-5ee8-49d1-ac49-8e62059e3cc1.jpg)

Although, notice that the model and log you are using for testing are such that the traces are already 100% compliant with the model (you are not going to see any deviation from the model visualization). You can download other examples to run from [here](https://doi.org/10.4121/uuid:a02afec8-b7c7-42b7-8dff-36d3de3032be), even though they are associated with our [ICAPS 2018 paper](https://www.aaai.org/ocs/index.php/ICAPS/ICAPS18/paper/viewPaper/17739) and, thus, consist in partially-ordered traces (that the plugin can handle).

**Comment by giacomolanciano** [comment]
~~In `dev`, there is an updated README where I clarify the actual requirements. I'm going to merge it in `master` as soon as I re-gain access to the SVN repo.~~
**EDIT 1:** the updated README is now merged in `master` and the SVN repo.

Indeed, my recommendation is to clone this repo and launch ProM from Eclipse (as explained in the README), instead of relying on ProM nightly builds. As of now, the TU/e server that hosts the release pipelines for many ProM packages seems down for maintenance and it is not clear when it will be up again.
**EDIT 2:** more info on how to configure Eclipse for ProM [here](https://svn.win.tue.nl/trac/prom/wiki/GettingStarted).

**Comment by matheusPereiraAlmeida** [comment]
Hi @giacomolanciano,  I followed yours instructions and used the logs from the link and downloaded python 2.7 and I was able to use the plugin from ProM, thank you!

![image](https://user-images.githubusercontent.com/61329801/143722948-c5cbcaf2-9b6e-4d2a-ad40-48b382668d99.png)

Later I'll try to use directly on Eclipse.

**Comment by matheusPereiraAlmeida** [comment]
@giacomolanciano , one quick question. If I clone this repo and open it with Eclipse, will I be able to debug the plugin?

**Comment by giacomolanciano** [comment]
> If I clone this repo and open it with Eclipse, will I be able to debug the plugin? 

Exactly, @matheusPereiraAlmeida. The advantage is that starting ProM from Eclipse you can see the log messages as the plug-ins are executed. That's how I discovered the exception thrown by the FD process. This way you could also apply and test your changes, if any.
